python/pyspark/sql/column.py

```diff
@@ -690,17 +798,22 @@ def between(self, lowerBound, upperBound):
         """
         return (self >= lowerBound) & (self <= upperBound)
 
-    @since(1.4)
     def when(self, condition, value):
         """
         Evaluates a list of conditions and returns one of multiple possible result expressions.
         If :func:`Column.otherwise` is not invoked, None is returned for unmatched conditions.
 
-        See :func:`pyspark.sql.functions.when` for example usage.
+        .. versionadded:: 1.4.0
 
-        :param condition: a boolean :class:`Column` expression.
-        :param value: a literal value, or a :class:`Column` expression.
+        Parameters
+        ----------
+        condition : :class:`Column`
+            a boolean :class:`Column` expression.
+        value :
```

Comment:
I simply make it un-typed for simplicity where it takes a column + any value that a literal supports.